applications/ChimeraApplication/custom_processes/calculate_signed_distance_to_2d_condition_skin_process.h

```diff
@@ -0,0 +1,2682 @@
+                    {
+                        if ((i != min_pos) && (i != max_pos))
+                        {
+                            center_pos = i;
+                        }
+                    }
+
+                    // ######## ADDING NEW NODE #########
+                    Node<3>::Pointer pnode1 = mrNewSkinModelPart.CreateNewNode(id_node++, edge_points[0].X(), edge_points[0].Y(), edge_points[0].Z());
+                    Node<3>::Pointer pnode2 = mrNewSkinModelPart.CreateNewNode(id_node++, edge_points[min_pos].X(), edge_points[min_pos].Y(), edge_points[min_pos].Z());
+                    Node<3>::Pointer pnode3 = mrNewSkinModelPart.CreateNewNode(id_node++, edge_points[center_pos].X(), edge_points[center_pos].Y(), edge_points[center_pos].Z());
+                    Node<3>::Pointer pnode4 = mrNewSkinModelPart.CreateNewNode(id_node++, edge_points[max_pos].X(), edge_points[max_pos].Y(), edge_points[max_pos].Z());
+
+                    // ######## ADDING NEW CONDITION #########
+                    //form two triangles
+                    Triangle3D3<Node<3>> triangle1(pnode1, pnode2, pnode3);
+                    Triangle3D3<Node<3>> triangle2(pnode1, pnode3, pnode4);
+
+                    Condition const &rReferenceCondition = KratosComponents<Condition>::Get("Condition3D");
+
+                    Properties::Pointer properties = mrNewSkinModelPart.rProperties()(0);
+
+                    Condition::Pointer p_condition1 = rReferenceCondition.Create(id_condition++, triangle1, properties);
+                    Condition::Pointer p_condition2 = rReferenceCondition.Create(id_condition++, triangle2, properties);
+
+                    mrNewSkinModelPart.Conditions().push_back(p_condition1);
+                    mrNewSkinModelPart.Conditions().push_back(p_condition2);
+                }*/
+            }
+        }
+    }
+
+    ///******************************************************************************************************************
+    ///******************************************************************************************************************
+
+    void GenerateQuadtree()
+    {
+        Timer::Start("Generating Quadtree");
+        //std::cout << "Generating the Quadtree..." << std::endl;
+        boost::shared_ptr<QuadtreeType> temp_quadtree = boost::shared_ptr<QuadtreeType>(new QuadtreeType());
+        //QuadtreeType::Pointer temp_quadtree = QuadtreeType::Pointer(new QuadtreeType() );
+        mpQuadtree.swap(temp_quadtree);
+
+        double low[2];
+        double high[2];
+
+        for (int i = 0; i < 2; i++)
+        {
+            low[i] = high[i] = mrFluidModelPart.NodesBegin()->Coordinates()[i];
+        }
+
+        // loop over all nodes in the bounding box
+        for (ModelPart::NodeIterator i_node = mrFluidModelPart.NodesBegin();
+             i_node != mrFluidModelPart.NodesEnd();
+             i_node++)
+        {
+            for (int i = 0; i < 2; i++)
+            {

```

Comment:
Try to use third party tool for this. It removes the burden from kratos maintainers.